test/lib/ufe/testCamera.py

```diff
@@ -234,13 +235,70 @@ def testUsdCamera(self):
         self._StartTest('TranslateRotate_vs_xform')
         mayaPathSegment = mayaUtils.createUfePathSegment('|stage|stageShape')
         cam2UsdPathSegment = usdUtils.createUfePathSegment('/cameras/cam2/camera2')
-        cam2Path = ufe.Path([mayaPathSegment, cam2UsdPathSegment])
-        cam2Item = ufe.Hierarchy.createItem(cam2Path)
+        camera2Path = ufe.Path([mayaPathSegment, cam2UsdPathSegment])
+        cam2Item = ufe.Hierarchy.createItem(camera2Path)
 
         cam2Camera = ufe.Camera.camera(cam2Item)
         cameraPrim = usdUtils.getPrimFromSceneItem(cam2Item)
         self._TestCameraAttributes(cam2Camera, cameraPrim)
 
+        # Requires updates to the cameraHandler interface from Ufe v4 or greater.
+        if ufeUtils.ufeFeatureSetVersion() < 4:
+            return
+        # Requires at least preview version 0.4.9
+        if (os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') < '4009'):
+            return
+
+        # Test that the camera handlers can find USD cameras in a scene segment
+        proxyShapeParentSegment = mayaUtils.createUfePathSegment('|stage')
+        camerasParentPathSegment = usdUtils.createUfePathSegment('/cameras')
+        camera2ParentPathSegment = usdUtils.createUfePathSegment('/cameras/cam2')
+        geoPathSegment = usdUtils.createUfePathSegment('/GEO')
+        camera1PathSegment = usdUtils.createUfePathSegment('/cameras/cam1/camera1')
+        
+        proxyShapePath = ufe.Path([mayaPathSegment])
+        proxyShapeParentPath = ufe.Path([proxyShapeParentSegment])
+        camerasParentPath = ufe.Path([mayaPathSegment, camerasParentPathSegment])
+        camera2ParentPath = ufe.Path([mayaPathSegment, camera2ParentPathSegment])
+        geoPath = ufe.Path([mayaPathSegment, geoPathSegment])
+        camera1Path = ufe.Path([mayaPathSegment, camera1PathSegment])
+
+        proxyShapeItem = ufe.Hierarchy.createItem(proxyShapePath)
+        proxyShapeParentItem = ufe.Hierarchy.createItem(proxyShapeParentPath)
+        camerasParentItem = ufe.Hierarchy.createItem(camerasParentPath)
+        camera2ParentItem = ufe.Hierarchy.createItem(camera2ParentPath)
+        geoItem = ufe.Hierarchy.createItem(geoPath)
+
+        # searching on a gateway item should give all cameras in the scene segment
+        # this will test ProxyShapeCameraHandler
+        result = ufe.CameraHandler.findAll(proxyShapeItem)
+        self.assertTrue(result.contains(camera1Path))
+        self.assertTrue(result.contains(camera2Path))
+        self.assertEqual(len(result), 2)
+
+        # searching the the parent of a gateway item searches the Maya scene segment
+        # for cameras and searches nested scene segments.
+        result = ufe.CameraHandler.findAll(proxyShapeParentItem)
+        self.assertTrue(result.contains(camera1Path))
+        self.assertTrue(result.contains(camera2Path))
+        self.assertEqual(len(result), 2)
```

Comment:
The Maya camera handler is not implemented, and there are no cameras that are children of proxyShapeParentItem. If we had the Maya camera handler and we had searched the Maya root path then we'd find front, side, top and persp.

Reply:
Right, so do we plan on leaving the Maya camera handler unimplemented, because there is no demonstrated need for it?  If so, would be nice to add a comment to that effect in the test.

Reply:
Sure I'll add a comment.